### PR TITLE
Base package for class path must be list property

### PR DESCRIPTION
**Pull request by kunsingh**
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/797)

### CODE REVIEW

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java [hunk 1]
```diff
@@ -41,21 +40,36 @@
...
+
+        //logging
         ConfigurationBuilder configuration = ConfigurationBuilder.build(DefaultClasspathScanner.class.getClassLoader());
         configuration = configuration.setExpandSuperTypes(false);
 
-        if(rootPackage != null && !rootPackage.trim().isEmpty()) {
-            configuration = configuration.forPackages(rootPackage);
-            configuration = configuration.setUrls(ClasspathHelper.forPackage(rootPackage));
-            configuration = configuration.filterInputsBy(new FilterBuilder().includePackage(rootPackage));
+        if(rootPackages != null && rootPackages.size() > 0) {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: `rootPackages != null` can be removed

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java [hunk 2]
```diff
@@ -41,21 +40,36 @@
...
+    public DefaultClasspathScanner(final String... rootPackages){
+        this(Arrays.asList(rootPackages));
     }
 
-    public DefaultClasspathScanner(final String rootPackage) {
+    public DefaultClasspathScanner(final List<String> rootPackages) {
+
+        Assert.requireNonNull(rootPackages, "rootPackages");
 
+        LOG.info("Scanning class path for root packages {}", Arrays.toString(rootPackages.toArray()));
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: Change log level to debug or trace

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java [hunk 3]
```diff
@@ -76,9 +90,9 @@ public DefaultClasspathScanner(final String rootPackage) {
             }
         }
         for (URL url : toRemove) {
+            LOG.info("Url removed {}", url.toString());
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: Change log level to trace

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java [hunk 4]
```diff
@@ -76,9 +90,9 @@ public DefaultClasspathScanner(final String rootPackage) {
...
             configuration.getUrls().remove(url);
         }
-
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: Log configuration.getUrls() in trace to see all URLS

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java [hunk 5]
```diff
@@ -48,21 +49,38 @@ public void testSimpleScan() {
     @Test
     public void testInPackageScan() {
         //There can't be a class that is annotated with Inject
-        DefaultClasspathScanner scanner = new DefaultClasspathScanner("com.canoo.impl");
-        Set<Class<?>> classes = scanner.getTypesAnnotatedWith(Resources.class);
-        assertNotNull(classes);
-        assertEquals(classes.size(), 0);
+        DefaultClasspathScanner scanner = new DefaultClasspathScanner("com.canoo.impl.server.classpathscan");
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: add final

Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: "com.canoo.impl.server.classpathscan" <- static field

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java [hunk 6]
```diff
@@ -48,21 +49,38 @@ public void testSimpleScan() {
...
+    }
 
-        classes = scanner.getTypesAnnotatedWith(AnnotationForClasspathScanTest.class);
-        assertNotNull(classes);
-        assertEquals(classes.size(), 1);
-        assertTrue(classes.contains(AnnotatedClassForClasspathScan.class));
+    @Test
+    public void testInMultiplePackages() {
+        //There can't be a class that is annotated with Inject
+        DefaultClasspathScanner scanner = new DefaultClasspathScanner("com.canoo.impl.server.classpathscan.documented", "com.canoo.impl.server.classpathscan.resource");
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: final & static field for strings

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java [hunk 7]
```diff
@@ -48,21 +49,38 @@ public void testSimpleScan() {
...
+        assertForAnnotation(scanner);
+    }
+
+    private void assertForAnnotation(DefaultClasspathScanner scanner) {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: final

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java [hunk 8]
```diff
@@ -48,21 +49,38 @@ public void testSimpleScan() {
...
+        Set<Class<?>> resourceClasses = scanner.getTypesAnnotatedWith(Resource.class);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: assert not null for scanner

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java [hunk 9]
```diff
@@ -48,21 +49,38 @@ public void testSimpleScan() {
...
+        assertNotNull(resourceClasses);
+        assertEquals(resourceClasses.size(), 1);
+
+        Set<Class<?>> documentClasses = scanner.getTypesAnnotatedWith(Documented.class);
+        assertNotNull(documentClasses);
+        assertEquals(documentClasses.size(), 1);
+    }
+
+    @Test
+    public void testInMultiplePackages2() {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Better name for method, not 2 ;)

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java [hunk 10]
```diff
@@ -36,7 +35,7 @@ public void testConfigLoad() {
 
             //then:
             assertEquals(configuration.getBooleanProperty(USE_CROSS_SITE_ORIGIN_FILTER), false);
-            assertEquals(configuration.getProperty(ROOT_PACKAGE_FOR_CLASSPATH_SCAN, ROOT_PACKAGE_FOR_CLASSPATH_SCAN_DEFAULT_VALUE), null);
+            assertEquals(configuration.getProperty(ROOT_PACKAGE_FOR_CLASSPATH_SCAN), null);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: I think we have a` getProperty(key)` that returns null if it is not found, so default value is not needed

**Comment by hendrikebbers** [comment]
fix for https://github.com/canoo/dolphin-platform/issues/726

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 10 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: its the assert against null. Default value is removed.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java (hunk 9 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Done.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java (hunk 8 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Done.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java (hunk 7 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Done.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java (hunk 6 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Done.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java (hunk 5 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Done.

Comment (kunsingh) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ClasspathScannerTest.java]: Done.

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java (hunk 3 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: Done.

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java (hunk 4 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: Done.

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java (hunk 1 above)
Comment (kunsingh) [reply platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/scanner/DefaultClasspathScanner.java]: Done.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 10 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: lol, you are right :)

**Pull request merged**